### PR TITLE
LPS-85586 Workflow custom actions :: Assign to me

### PR DESCRIPTION
**Pull request by dgarciasarai**

### CODE REVIEW

File: modules/apps/headless-apio/portal/portal-apio-api/src/main/java/com/liferay/portal/apio/permission/HasPermissionUtil.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+public class HasPermissionUtil {
+
+	/**
+	 * Execute the received function and transforms any occurred exception into a {@code false}
+	 */
+	public static BiFunction<Credentials, Long, Boolean> failOnException(
+		ThrowableBiFunction<Credentials, Long, Boolean> throwableBiFunction) {
+
+		return (credentials, aLong) -> Try.fromFallible(
```
Comment (nhpatt) [inline modules/apps/headless-apio/portal/portal-apio-api/src/main/java/com/liferay/portal/apio/permission/HasPermissionUtil.java]: Can you rename to id?

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java [hunk 2]
```diff
@@ -98,14 +101,17 @@ public String getName() {
...
 			AcceptLanguage.class, DDMFormRenderingContext.class,
 			ThemeDisplay.class, FormContextIdentifier.class,
-			this::_hasPermission, FormContextForm::buildForm
+			failOnException(_hasPermission.forAddingIn(
+				FormInstanceRecordIdentifier.class)), FormContextForm::buildForm
 		).addCustomRoute(
 			new FetchLatestDraftRoute(), this::_fetchDDMFormInstanceRecord,
 			CurrentUser.class, FormInstanceRecordIdentifier.class,
-			this::_hasPermission, FetchLatestDraftForm::buildForm
+			failOnException(_hasPermission.forAddingIn(
```
Comment (nhpatt) [inline modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java]: It's the same method call, right?

File: modules/apps/headless-apio/portal/portal-apio-api/src/main/java/com/liferay/portal/apio/permission/HasPermissionUtil.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * </p>
+ *
+ * @author Sarai Díaz
+ * @review
+ */
+public class HasPermissionUtil {
+
+	/**
+	 * Execute the received function and transforms any occurred exception into a {@code false}
+	 */
```
Comment (nhpatt) [inline modules/apps/headless-apio/portal/portal-apio-api/src/main/java/com/liferay/portal/apio/permission/HasPermissionUtil.java]: @review

File: modules/apps/headless-apio/workflow/workflow-apio-impl/src/main/java/com/liferay/workflow/apio/internal/architect/form/AssignToMeForm.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		Form.Builder<AssignToMeForm> formBuilder) {
+
+		return formBuilder.title(
+			__ -> "Assign task"
+		).description(
+			__ ->
+				"This form can be used to assign a task to an user"
+		).constructor(
+			AssignToMeForm::new
+		).addOptionalString(
```
Comment (nhpatt) [inline modules/apps/headless-apio/workflow/workflow-apio-impl/src/main/java/com/liferay/workflow/apio/internal/architect/form/AssignToMeForm.java]: sort

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java (hunk 2 above)
Comment (dgarciasarai) [reply modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java]: yes, it's the same but, using the permission util function `failOnException` will return false if there is an exception

**Pull request merged**